### ISSUE TITLE
tests fail with Rust 1.71+

### ISSUE THREAD

**Issue by decathorpe** [Lucretiel/cool_asserts#5]
We're packaging the cool_asserts crate for Fedora Linux as a dependency of nom-supreme and other crates, and run tests during the build process. I noticed that doctests have started to fail with Rust 1.71.0 - this is also reproducible by checking out this git repo and just running "cargo test --doc".

The error messages don't make much sense to me, but it looks like the line numbers in the error message are different from the expected ones with Rust 1.71:

```
$ cargo test --doc
    Finished test [unoptimized + debuginfo] target(s) in 0.00s
   Doc-tests cool_asserts

running 16 tests
test src/assert_panics.rs - assert_panics::assert_panics (line 21) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 30) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 109) ... ok
test src/assert_panics.rs - assert_panics::get_panic_message (line 347) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 81) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 58) ... ok
test src/assert_matches.rs - assert_matches::assert_matches (line 72) ... ok
test src/assert_matches.rs - assert_matches::assert_matches (line 49) ... ok
test src/assertion_failure.rs - assertion_failure::assertion_failure (line 88) ... FAILED
test src/assert_matches.rs - assert_matches::assert_matches (line 109) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 47) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 117) ... ok
test src/assert_matches.rs - assert_matches::assert_matches (line 146) ... ok
test src/assert_panics.rs - assert_panics::assert_panics (line 88) ... ok
test src/assert_matches.rs - assert_matches::assert_matches (line 85) ... ok
test src/assert_matches.rs - assert_matches::assert_matches (line 129) ... ok

failures:

---- src/assertion_failure.rs - assertion_failure::assertion_failure (line 88) stdout ----
Test executable failed (exit status: 101).

stderr:
thread 'main' panicked at 'assertion failed at src/assertion_failure.rs:8: `(Assertion Message)`
      key: 10
 long_key: "Hello\tWorld!"
  Trailing message: Hello, World!', src/assertion_failure.rs:8:30
stack backtrace:
   0:     0x564f5d9afbf1 - std::backtrace_rs::backtrace::libunwind::trace::h6aeaf83abc038fe6
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x564f5d9afbf1 - std::backtrace_rs::backtrace::trace_unsynchronized::h4f9875212db0ad97
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x564f5d9afbf1 - std::sys_common::backtrace::_print_fmt::h3f820027e9c39d3b
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x564f5d9afbf1 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hded4932df41373b3
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x564f5d9cb5ff - core::fmt::rt::Argument::fmt::hc8ead7746b2406d6
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/fmt/rt.rs:138:9
   5:     0x564f5d9cb5ff - core::fmt::write::hb1cb56105a082ad9
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/fmt/mod.rs:1094:21
   6:     0x564f5d9adf11 - std::io::Write::write_fmt::h797fda7085c97e57
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/io/mod.rs:1713:15
   7:     0x564f5d9afa05 - std::sys_common::backtrace::_print::h492d3c92d7400346
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x564f5d9afa05 - std::sys_common::backtrace::print::hf74aa2eef05af215
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x564f5d9b0d87 - std::panicking::default_hook::{{closure}}::h8cad394227ea3de8
  10:     0x564f5d9b0b74 - std::panicking::default_hook::h249cc184fec99a8a
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:288:9
  11:     0x564f5d9b123c - std::panicking::rust_panic_with_hook::h82ebcd5d5ed2fad4
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:705:13
  12:     0x564f5d9b1137 - std::panicking::begin_panic_handler::{{closure}}::h810bed8ecbe66f1a
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:597:13
  13:     0x564f5d9b0026 - std::sys_common::backtrace::__rust_end_short_backtrace::h1410008071796261
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:151:18
  14:     0x564f5d9b0e82 - rust_begin_unwind
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:593:5
  15:     0x564f5d995ed3 - core::panicking::panic_fmt::ha0a42a25e0cf258d
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/panicking.rs:67:14
  16:     0x564f5d998381 - rust_out::main::_doctest_main_src_assertion_failure_rs_88_0::{{closure}}::h5c55905a5925f0e4
  17:     0x564f5d996856 - std::panicking::try::do_call::hf9268c9e896f9d85
  18:     0x564f5d9983ab - __rust_try
  19:     0x564f5d99680c - std::panicking::try::h8697149df0d24bc8
  20:     0x564f5d9967e6 - std::panic::catch_unwind::h851899c16c918a93
  21:     0x564f5d997c7c - rust_out::main::_doctest_main_src_assertion_failure_rs_88_0::h4877663af0579826
  22:     0x564f5d997c66 - rust_out::main::h5cfeb40e49780147
  23:     0x564f5d996ba3 - core::ops::function::FnOnce::call_once::hffc7fce7ac910f2d
  24:     0x564f5d996776 - std::sys_common::backtrace::__rust_begin_short_backtrace::hb65aab94565099a4
  25:     0x564f5d9967c9 - std::rt::lang_start::{{closure}}::h6016435505c8a5a2
  26:     0x564f5d9abd85 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h5ce27e764c284c0a
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/ops/function.rs:284:13
  27:     0x564f5d9abd85 - std::panicking::try::do_call::h4c1fc390ae241991
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:500:40
  28:     0x564f5d9abd85 - std::panicking::try::h4d36e7eaed86af72
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:464:19
  29:     0x564f5d9abd85 - std::panic::catch_unwind::h41cfb4dd65282b1e
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panic.rs:142:14
  30:     0x564f5d9abd85 - std::rt::lang_start_internal::{{closure}}::hfed411c1c5fdb925
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/rt.rs:148:48
  31:     0x564f5d9abd85 - std::panicking::try::do_call::h6893f6f32a464342
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:500:40
  32:     0x564f5d9abd85 - std::panicking::try::h52b7102f469a0567
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:464:19
  33:     0x564f5d9abd85 - std::panic::catch_unwind::h62120054677916b5
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panic.rs:142:14
  34:     0x564f5d9abd85 - std::rt::lang_start_internal::hd66bf6b7da144005
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/rt.rs:148:20
  35:     0x564f5d9967a7 - std::rt::lang_start::hf20dd7c855f889aa
  36:     0x564f5d9983e5 - main
  37:     0x7fcdf01c9b4a - __libc_start_call_main
  38:     0x7fcdf01c9c0b - __libc_start_main@@GLIBC_2.34
  39:     0x564f5d996535 - _start
  40:                0x0 - <unknown>
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"assertion failed at src/assertion_failure.rs:8: `(Assertion Message)`\n      key: 10\n long_key: \"Hello\\tWorld!\"\n  Trailing message: Hello, World!"`,
 right: `"assertion failed at src/assertion_failure.rs:7: `(Assertion Message)`\n      key: 10\n long_key: \"Hello\\tWorld!\"\n  Trailing message: Hello, World!"`', src/assertion_failure.rs:23:1
stack backtrace:
   0:     0x564f5d9afbf1 - std::backtrace_rs::backtrace::libunwind::trace::h6aeaf83abc038fe6
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x564f5d9afbf1 - std::backtrace_rs::backtrace::trace_unsynchronized::h4f9875212db0ad97
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x564f5d9afbf1 - std::sys_common::backtrace::_print_fmt::h3f820027e9c39d3b
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x564f5d9afbf1 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hded4932df41373b3
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x564f5d9cb5ff - core::fmt::rt::Argument::fmt::hc8ead7746b2406d6
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/fmt/rt.rs:138:9
   5:     0x564f5d9cb5ff - core::fmt::write::hb1cb56105a082ad9
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/fmt/mod.rs:1094:21
   6:     0x564f5d9adf11 - std::io::Write::write_fmt::h797fda7085c97e57
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/io/mod.rs:1713:15
   7:     0x564f5d9afa05 - std::sys_common::backtrace::_print::h492d3c92d7400346
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x564f5d9afa05 - std::sys_common::backtrace::print::hf74aa2eef05af215
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x564f5d9b0d87 - std::panicking::default_hook::{{closure}}::h8cad394227ea3de8
  10:     0x564f5d9b0b74 - std::panicking::default_hook::h249cc184fec99a8a
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:288:9
  11:     0x564f5d9b123c - std::panicking::rust_panic_with_hook::h82ebcd5d5ed2fad4
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:705:13
  12:     0x564f5d9b1137 - std::panicking::begin_panic_handler::{{closure}}::h810bed8ecbe66f1a
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:597:13
  13:     0x564f5d9b0026 - std::sys_common::backtrace::__rust_end_short_backtrace::h1410008071796261
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/sys_common/backtrace.rs:151:18
  14:     0x564f5d9b0e82 - rust_begin_unwind
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:593:5
  15:     0x564f5d995ed3 - core::panicking::panic_fmt::ha0a42a25e0cf258d
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/panicking.rs:67:14
  16:     0x564f5d99613b - core::panicking::assert_failed_inner::h8de29b2c8e31440f
  17:     0x564f5d997186 - core::panicking::assert_failed::hc7f75ded9abb0590
  18:     0x564f5d997d88 - rust_out::main::_doctest_main_src_assertion_failure_rs_88_0::h4877663af0579826
  19:     0x564f5d997c66 - rust_out::main::h5cfeb40e49780147
  20:     0x564f5d996ba3 - core::ops::function::FnOnce::call_once::hffc7fce7ac910f2d
  21:     0x564f5d996776 - std::sys_common::backtrace::__rust_begin_short_backtrace::hb65aab94565099a4
  22:     0x564f5d9967c9 - std::rt::lang_start::{{closure}}::h6016435505c8a5a2
  23:     0x564f5d9abd85 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h5ce27e764c284c0a
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/ops/function.rs:284:13
  24:     0x564f5d9abd85 - std::panicking::try::do_call::h4c1fc390ae241991
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:500:40
  25:     0x564f5d9abd85 - std::panicking::try::h4d36e7eaed86af72
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:464:19
  26:     0x564f5d9abd85 - std::panic::catch_unwind::h41cfb4dd65282b1e
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panic.rs:142:14
  27:     0x564f5d9abd85 - std::rt::lang_start_internal::{{closure}}::hfed411c1c5fdb925
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/rt.rs:148:48
  28:     0x564f5d9abd85 - std::panicking::try::do_call::h6893f6f32a464342
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:500:40
  29:     0x564f5d9abd85 - std::panicking::try::h52b7102f469a0567
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:464:19
  30:     0x564f5d9abd85 - std::panic::catch_unwind::h62120054677916b5
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panic.rs:142:14
  31:     0x564f5d9abd85 - std::rt::lang_start_internal::hd66bf6b7da144005
                               at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/rt.rs:148:20
  32:     0x564f5d9967a7 - std::rt::lang_start::hf20dd7c855f889aa
  33:     0x564f5d9983e5 - main
  34:     0x7fcdf01c9b4a - __libc_start_call_main
  35:     0x7fcdf01c9c0b - __libc_start_main@@GLIBC_2.34
  36:     0x564f5d996535 - _start
  37:                0x0 - <unknown>



failures:
    src/assertion_failure.rs - assertion_failure::assertion_failure (line 88)

test result: FAILED. 15 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.25s

error: doctest failed, to rerun pass `--doc`
```

**Comment by decathorpe** [comment]
Looks like the underlying cause was found on the rust-lang zulip: https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/doctest.20line.20numbers/near/378952644

TL;DR: rustdoc 1.71 now inserts one more line in the "preamble" for rustdoc tests, skewing the line numbers by +1 compared to Rust <= 1.70.

**Comment by michel-slm** [comment]
This is fixed in https://github.com/Lucretiel/cool_asserts/commit/ef7189c05f287603baa27e44af61aecaf692c414 I think. Tests pass with 2.0.4
